### PR TITLE
Fix for git tags

### LINKED ISSUE #6821: [BUG] using git tags in `mlflow run --version TAG` fails
### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

1.28.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS 12.4
- **Python version**: 3.9.13
- **yarn version, if running the dev UI**: None

### Describe the problem

First, thanks a lot for this project.

In the command `mlflow run --version VVV https://...`, the term _version_ looks like you could use a git tag, and a few versions ago (I think around 1.14) this used to work.

However, in recent versions such as 1.27 and 1.28.0, using an existing git tag such as `1.0.0` just fails with:

```
2022/09/17 19:16:52 ERROR mlflow.cli: === Unable to checkout version '1.0.0' of git repo https://github.com/org/repo- please ensure that the version exists in the repo. Error: Cmd('git') failed due to: exit code(1)
  cmdline: git checkout 1.0.0
  stderr: 'error: pathspec '1.0.0' did not match any file(s) known to git' ===
```

The command only works with branch names or commit SHAs. Is that the desired behavior? The docs for this `--version` argument mention "Git commit references", so it looks like tags should be accepted.

Ideally, could you please reinstate this feature?
Thanks!

### Tracking information

_No response_

### Code to reproduce issue

```console
$ pip install mlflow==1.28.0
$ mlflow run git@github.com:graelo/mlflow-example.git -v 1.0.0
2022/09/17 19:26:55 INFO mlflow.projects.utils: === Fetching project from git@github.com:graelo/mlflow-example.git into /var/folders/hb/9_9_kvm96fb7w3zrqwzp3kkh0000gn/T/tmpesd_pz9h ===
2022/09/17 19:26:57 ERROR mlflow.cli: === Unable to checkout version '1.0.0' of git repo git@github.com:graelo/mlflow-example.git- please ensure that the version exists in the repo. Error: Cmd('git') failed due to: exit code(1)
  cmdline: git checkout 1.0.0
  stderr: 'error: pathspec '1.0.0' did not match any file(s) known to git' ===
```

### Stack trace

```console
$ mlflow run git@github.com:graelo/mlflow-example.git -v 1.0.0
2022/09/17 19:26:55 INFO mlflow.projects.utils: === Fetching project from git@github.com:graelo/mlflow-example.git into /var/folders/hb/9_9_kvm96fb7w3zrqwzp3kkh0000gn/T/tmpesd_pz9h ===
2022/09/17 19:26:57 ERROR mlflow.cli: === Unable to checkout version '1.0.0' of git repo git@github.com:graelo/mlflow-example.git- please ensure that the version exists in the repo. Error: Cmd('git') failed due to: exit code(1)
  cmdline: git checkout 1.0.0
  stderr: 'error: pathspec '1.0.0' did not match any file(s) known to git' ===
```

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [X] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by AvikantSrivastava**
Signed-off-by: avikantsrivastava <asdavikant@gmail.com>

## Related Issues/PRs

Fix https://github.com/mlflow/mlflow/issues/6821

## What changes are proposed in this pull request?

Git tags were not being fetched by default during a git fetch operation here https://github.com/mlflow/mlflow/blob/5ddfe9a1b48e065540094d83125040d3273c48fa/mlflow/projects/utils.py#L211

I have added a command to explicitly fetch the tags 
```git fetch --tags```

## How is this patch tested?

- [] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

I ran the example project with the tags, which was earlier failing (as mentioned in the issue https://github.com/mlflow/mlflow/issues/6821)

With the fix, it is working as expected, attaching logs here
```bash
mlflow run git@github.com:graelo/mlflow-example.git -v "1.0.0"
```
```
(mlflow-dev) (base) aesthetixell@aesthetixell-pc:~/Desktop/Working/open_source/mlflow$ mlflow run git@github.com:graelo/mlflow-example.git -v "1.0.0"
2022/10/13 12:42:42 INFO mlflow.projects.utils: === Fetching project from git@github.com:graelo/mlflow-example.git into /tmp/tmpl0su6459 ===
2022/10/13 12:42:49 INFO mlflow.utils.conda: Conda environment mlflow-2b6b69a3ea30872171ff71aee564367746fae613 already exists.
2022/10/13 12:42:49 INFO mlflow.projects.utils: === Created directory /tmp/tmphvt0alup for downloading remote URIs passed to arguments of type 'path' ===
2022/10/13 12:42:49 INFO mlflow.projects.backend.local: === Running command 'source /home/aesthetixell/miniconda3/bin/../etc/profile.d/conda.sh && conda activate mlflow-2b6b69a3ea30872171ff71aee564367746fae613 1>&2 && python train.py 0.5 0.1' in run with ID 'bf4deb8de7954652abbac9df1d78dc63' === 
Elasticnet model (alpha=0.500000, l1_ratio=0.100000):
  RMSE: 0.7947931019036528
  MAE: 0.6189130834228137
  R2: 0.1841166871822183
2022/10/13 12:42:53 INFO mlflow.projects: === Run (ID 'bf4deb8de7954652abbac9df1d78dc63') succeeded ===
```

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [x] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by AvikantSrivastava** [comment]
Hey @BenWilson2 @graelo
the maintainer need to trigger CI workflows,
can you look into it?

File: mlflow/projects/utils.py [hunk 1]
```diff
@@ -210,6 +210,9 @@ def _fetch_git_repo(uri, version, dst_dir):
     if version is not None:
         try:
             origin.fetch(refspec=version, depth=GIT_FETCH_DEPTH)
```
Comment (harupy) [inline mlflow/projects/utils.py:212]:
```suggestion
            origin.fetch(refspec=version, depth=GIT_FETCH_DEPTH, tags=True)
```

Does this work?

**Comment by graelo** [comment]
Thanks a lot!

File: mlflow/projects/utils.py (hunk 1 above)
Comment (AvikantSrivastava) [reply mlflow/projects/utils.py:212]: yes, it works

Comment (AvikantSrivastava) [reply mlflow/projects/utils.py]: @harupy checked it on my local and made the change, can you review it?

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by graelo** [comment]
Thanks @AvikantSrivastava!

**Pull request merged**